core/trino-main/src/main/java/io/trino/server/protocol/JsonEncodingUtils.java

```diff
@@ -369,13 +368,9 @@ public void encode(JsonGenerator generator, ConnectorSession session, Block bloc
             generator.writeStartObject();
             for (int i = 0; i < map.getSize(); i++) {
                 // Field name is always written as String for backward compatibility,
-                // (except for VarbinaryType where we encode it as Base64 representation of its bytes)
                 // and only value is properly encoded using it's type.
                 // TODO: improve in v2 JSON format
-                switch (mapType.getKeyType().getObjectValue(session, keyBlock, offset + i)) {
-                    case SqlVarbinary varbinary -> generator.writeFieldName(Base64.getEncoder().encodeToString(varbinary.getBytes()));
-                    case Object value -> generator.writeFieldName(value.toString());
-                }
+                generator.writeFieldName(mapType.getKeyType().getObjectValue(session, keyBlock, offset + i).toString());
```

Comment:
Comment above mentions backward compatibility. But we are changing how it works just know. Is that commment even important at all?

Reply:
It is, JSON v1 format always serializes key to a String, even if it is a map or list.